libandroid-navigation/src/main/java/com/mapbox/services/android/navigation/v5/navigation/metrics/NavigationMetricsWrapper.java

```diff
@@ -221,17 +221,17 @@ private static void updateRouteProgressSessionData(MetricsRouteProgress routePro
     // Check if location update happened before or after the reroute
     List<Location> afterLoc = new ArrayList<>();
 
-    if (sessionState.afterRerouteLocations() != null) {
-      for (Location loc : sessionState.afterRerouteLocations()) {
-        if (loc.getTime() > sessionState.rerouteDate().getTime()) {
+    if (sessionState.afterEventLocations() != null) {
+      for (Location loc : sessionState.afterEventLocations()) {
+        if (loc.getTime() > sessionState.eventDate().getTime()) {
           afterLoc.add(loc);
         }
       }
     }
 
     previousName = routeProgress.getCurrentStepName();
 
-    beforeLocations = obtainLocations(sessionState.beforeRerouteLocations());
+    beforeLocations = obtainLocations(sessionState.beforeEventLocations());
```

Comment:
Are these methods (including the afterEventLocations() above) still necessary with the dynamic creation of location arrays from the location buffer?

Reply:
Yeah good catch - all before / after location logic is now in `NavigationTelemetry`